src/rpc/multipass.proto

```diff
@@ -201,30 +207,55 @@ message InstanceStatus {
     Status status = 1;
 }
 
-message InfoReply {
-    message Info {
-        string name = 1;
-        InstanceStatus instance_status = 2;
-        string image_release = 3;
-        string current_release = 4;
-        string id = 5;
-        string load = 6;
-        string memory_usage = 7;
-        string memory_total = 8;
-        string disk_usage = 9;
-        string disk_total = 10;
-        repeated string ipv4 = 11;
-        repeated string ipv6 = 12;
-        MountInfo mount_info = 13;
-        string cpu_count = 14;
+message InstanceDetails {
+    string image_release = 1;
+    string current_release = 2;
+    string id = 3;
+    string load = 4;
+    string memory_usage = 5;
+    string disk_usage = 6;
+    repeated string ipv4 = 7;
+    repeated string ipv6 = 8;
+    int32 num_snapshots = 9;
+}
+
+message SnapshotFundamentals {
+    string snapshot_name = 1;
+    string parent = 2;
+    string comment = 3;
+    google.protobuf.Timestamp creation_timestamp = 4;
+}
+
+message SnapshotDetails {
+    SnapshotFundamentals fundamentals = 1;
+    string size = 2;
+    repeated string children = 4;
+}
+
+message DetailedInfoItem {
+    string name = 1;
+    InstanceStatus instance_status = 2;
+    string memory_total = 3;
+    string disk_total = 4;
+    string cpu_count = 5;
+    MountInfo mount_info = 6;
+
+    oneof extra_info {
+        InstanceDetails instance_info = 7;
+        SnapshotDetails snapshot_info = 8;
     }
-    repeated Info info = 1;
-    string log_line = 2;
+}
+
+message InfoReply {
+    repeated DetailedInfoItem details = 1;
+    bool snapshots = 2; // useful to determine what entity (instance/snapshot) was absent when details are empty
+    string log_line = 3;
```

Comment:
OK. There are other cases where it is not the first and `InfoReply` was already one of them. Can you confirm if you are looking to get this fixed in cases where `log_line` was touched at all, but not others, in this PR?

Reply:
Well, in the end, it doesn't really matter as we are not guaranteeing not breaking API, but just a preference by me to have the same types defined first for each message type and then the more unique types after that.  I think this will help in the long run if and we do guarantee API compatibility.

For now and the sake of getting this done, we can just leave these as is.  As an aside, we should clean up our API come time to get us to API stability.  I will resolve all of the comments surrounding this topic.